extensions/wikia/Discussions/maintenance/SiteStatus.class.php

```diff
@@ -0,0 +1,411 @@
+			->WHERE( 'site_id' )->EQUAL_TO( $this->siteId )
+			->run(
+				$this->getCentralDbr(),
+				function( $result ) {
+					/** @var \ResultWrapper|bool $result */
+					if ( !is_object( $result ) ) {
+						return;
+					}
+
+					$row = $result->fetchObject();
+
+					if ( empty( $row ) ) {
+						return;
+					}
+
+					$this->hasStatusRecord = true;
+					$this->status = $row->status;
+					$this->siteAvailable = $row->site_available;
+					$this->discussionsEnabled = $row->discussions_enabled;
+					$this->navigationEnabled = $row->navigation_enabled;
+					$this->forumsEnabled = $row->forums_enabled;
+					$this->numThreadedForumPosts = $row->tf_posts;
+					$this->numWikiForumPosts = $row->wf_posts;
+					$this->discussionsFound = new \DateTime( $row->discussions_found );
+					$this->firstPostFound = new \DateTime( $row->first_post_found );
+					$this->lastMigrated = new \DateTime( $row->last_migrated );
+					$this->lastUpdated = new \DateTime( $row->last_updated );
+				}
+			);
+
+		$this->cityInfo = \WikiFactory::getWikiByDB( $this->dbName );
+	}
+
+	public function update() {
+		$this->probeSiteAvailable();
+		$this->updateWikiVariables();
+		$this->findExistingPosts();
+		$this->probeDiscussions();
+	}
+
+	public function writeBack() {
+		$this->debug( "Writing content to " . self::TABLE_STATUS );
+
+		$statement = new \WikiaSQL();
+		if ( $this->hasStatusRecord ) {
+			$statement
+				->UPDATE( self::TABLE_STATUS )
+				->WHERE( 'site_id' )->EQUAL_TO( $this->siteId );
+		} else {
+			$statement
+				->INSERT( self::TABLE_STATUS )
+				->SET( 'site_id', $this->siteId );
+		}
+
+		$now = (new \DateTime())->format('Y-m-d H:i:s');
+		$statement->SET( 'site_available', $this->siteAvailable )
+			->SET( 'discussions_enabled', $this->discussionsEnabled )
+			->SET( 'navigation_enabled', $this->navigationEnabled )
+			->SET( 'forums_enabled', $this->forumsEnabled )
+			->SET( 'tf_posts', $this->numThreadedForumPosts )
+			->SET( 'wf_posts', $this->numWikiForumPosts )
+			->SET( 'discussions_found', $this->discussionsFoundString() )
+			->SET( 'first_post_found', $this->firstPostFoundString() )
+			->SET( 'last_updated', $now )
+			->run( $this->getCentralDbw() );
+	}
+
+	public function discussionsFoundString() {
+		if ( empty( $this->discussionsFound ) ) {
+			return '';
+		}
+gmark("Setting: ".$this->discussionsFound->format('Y-m-d H:i:s'));
+		return $this->discussionsFound->format('Y-m-d H:i:s');
+	}
+
+	public function firstPostFoundString() {
+		if ( empty( $this->firstPostFound ) ) {
+			return '';
+		}
+		return $this->firstPostFound->format('Y-m-d H:i:s');
+	}
+
+	private function probeSiteAvailable() {
+		$this->debug( "Probing wiki site" );
+
+		$result = ( new \WikiaSQL() )
+			->SELECT( "city_flags" )
+			->FROM( self::TABLE_CITY_LIST )
+			->WHERE( 'city_id' )->EQUAL_TO( $this->siteId )
+				->AND_( 'city_public' )->EQUAL_TO( 1 )
+			->run(
+				$this->getCentralDbr(),
+				function(  $result ) {
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) Reduce the number of returns of this function 4, down to the maximum allowed 3. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://sonarqube.wikia-inc.com/coding_rules#rule_key=php%3AS1142)